### ISSUE TITLE
Prevent transitive references (i.e. references to references)

### ISSUE THREAD

**Issue by rugk** [keepassxreboot/keepassxc#2528]
[TIP]:  # ( Provide a general summary of the issue in the title above ^^ )
[TIP]:  # ( Do not include screenshots of your actual database! )

I guess when you reference references it may not result in good things…

## Steps to Reproduce
[NOTE]: # ( Provide a link to a live example, or an unambiguous set of steps to )
[NOTE]: # ( reproduce this bug. Include code to reproduce, if relevant )
1. Duplicate an entry.
2. Look at the ref ID, e.g. of the username. (Let's say it is "A")
3. Duplicate that entry again.
4. Now look at the ref ID (of the same, i.e. the username obviously).

## Expected Behavior
[NOTE]: # ( If you're describing a bug, tell us what should happen )
[NOTE]: # ( If you're suggesting a change/improvement, tell us how it should work )

Ref ID should be A.

## Current Behavior
[NOTE]: # ( If describing a bug, tell us what happens instead of the expected behavior )
[NOTE]: # ( If suggesting a change/improvement, explain the difference from the current behavior )

Ref ID is a new one (say "B"), so I assume this is the reference to the reference entry.

## Possible Problem
[NOTE]: # ( Not required, but suggest a fix/reason for the bug, )
[NOTE]: # ( or ideas how to implement the addition or change )

E.g. what happens if I delete the first reference?

## Context
[NOTE]: # ( How has this issue affected you? What are you trying to accomplish? )
[NOTE]: # ( Providing context helps us come up with a solution that is most useful in the real world )

Basically this came up when I tried these references, see https://github.com/keepassxreboot/keepassxc/issues/2527.

## Debug Info
[NOTE]: # ( Paste debug info from Help → About here )
KeePassXC - Version 2.3.4
Revision: 6fe821c

Bibliotheken:
- Qt 5.11.1
- libgcrypt 1.8.3

Betriebssystem: Fedora 29 (Workstation Edition)
CPU-Architektur: x86_64

**Comment by rugk** [comment]
Why closed? And why is this a new feature? It's rather a kind of a bug is not it? As said, if you delete the original reference of a reference, things likely break… this should be prevented…

**Comment by droidmonkey** [comment]
Why should it be prevented? It is perfectly reasonable and aligns with KeePass spec. If you choose to delete something in the reference chain, that is technically on you. Ideally the application would "heal" the reference for you.

**Comment by rugk** [comment]
> Ideally the application would "heal" the reference for you.

Okay, so if it does not do that, that would be a/the feature request (here), would not it?

**Comment by droidmonkey** [comment]
Sure, I can go with that.
